.github/workflows/test.yml

```diff
@@ -41,13 +35,20 @@ jobs:
         run: |
           # We want to run tests as a regular user, similar to Qubes VMs
           useradd --create-home --shell /bin/bash user
+      - name: Install Rust-specific dependencies
+        run: apt-get install --yes build-essential curl libssl-dev pkg-config
+        if: ${{ matrix.component == 'proxy' }}
+      - uses: actions/checkout@v4
+      # Install Rust 1.74.1, keep in sync with rust-toolchain.toml
+      - uses: dtolnay/rust-toolchain@1.74.1
```

Comment:
Flagging that this is the first use of a non "official GitHub" action - but 1) this is just CI, and 2) dtolnay is a trusted Rust project member and is trusted in our Rust audits as well.